server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -240,6 +294,27 @@ private void writeDefaultValue(StreamOutput out, Object defaultValue) throws IOE
         }
     }
 
+    private Object readParser(StreamInput in) throws IOException {
+        switch (type) {
+            case Boolean:
+                return new BooleanParser(in);
+            case Integer:
+                return new IntegerParser(in);
+            case Long:
+                return new LongParser(in);
+            case Float:
+                return new FloatParser(in);
+            case Double:
+                return new DoubleParser(in);
+            case TimeValue:
```

Comment:
This needs to match the implementation in `writeTo()`.  For this case you will read the boolean and if it's true do minmax, if false just do min.

Reply:
@dbwiddis since we are have `parser` in `readParser`, shouldn't the check match with `writeParser`. For ex:
instead of reading the boolean we can check the instance type same as read:
```
parser instanceof MinMaxTimeValueParser